Source/Concurrency/CivlTypeChecker.cs

```diff
@@ -1284,35 +1291,67 @@ private void VisitYieldingProcCallCmd(CallCmd call, YieldingProc callerProc, Yie
                 }
             }
 
-            private void VisitIntroductionProcCallCmd(CallCmd call, YieldingProc callerProc, IntroductionProc calleeProc)
+            private void VisitIntroductionProcCallCmd(CallCmd call, YieldingProc callerProc,
+                IntroductionProc calleeProc)
             {
                 if (!calleeProc.layerRange.Contains(callerProc.upperLayer))
```

Comment:
I think this first if block can be removed. It only applies to the case `!calleeProc.IsLemma` where the same check is already done.

Reply:
Done.